Sources/Core/ApplicationObserving.swift

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import UIKit
+
+/**
+ Type-erasure wrapper for `ApplicationObserving` to allow for implementing `Equatable`
+ which helps to avoid storing and handling duplicate observers
+ */
+struct AnyApplicationObserving: Equatable {
```

Comment:
I think we could use protocol ApplicationObserving directly, is there any reason for creating this struct?

Reply:
Yes. I ran into the error message `Protocol 'ApplicationObserving' can only be used as a generic constraint because it has Self or associated type requirements` which prevented me from storing an array of observers on the `FBApplicationDelegate`. 

Simplified example to post in a playground:
```
protocol Observing: Equatable {}

struct ObserverHolding {
    let observers: [Observing]
}
```

Reply:
Is adding `Equatable` worth the cost of adding the extra struct. I feel like the problems it solves might not be worth the cost.

Reply:
It's pretty necessary, otherwise there's nothing to prevent storing (and thereby calling) duplicate observers. Also there will be no way to remove an observer which seems like functionality we would want. I'm more concerned about the first issue.